### DISCUSSION TITLE
Unable to Find Function App with Multi-Subscription SPN

### DISCUSSION THREAD

**Discussion by nerddtvg** [Azure/functions-action#152] [Feature asks]
It seems when the service principal has multiple subscriptions available, the function app is not found if it is not in the default subscription.

The action should search all subscriptions, support specifying a subscription ID, or have the documentation state the limitation and workaround with `az acount set --subscription`.

**Comment by patelchandni** [comment]
@nerddtvg Did you create service principle using the recommended steps in [README](https://github.com/Azure/functions-action/blob/patelchandni/validation/README.md#using-azure-service-principal-for-rbac-as-deployment-credential )?

**Comment by nerddtvg** [comment]
@patelchandni These are existing service principals that have defined `subscriptionId` values in the JSON already as required by other tasks that also only search the default subscription. I can't change that value. Hence why I think it would be ideal if this task (and those others) allowed us to specify a subscription ID or search across all available subscriptions if that is provided. The simplest way is specifying the subscription as an optional parameter, otherwise searching across all subscriptions could end up with duplicates or ARM API request quota issues.

Or it could be mentioned in the documentation that using `az account set --subscription` as a command prior to running this task, you can specify something different. Just for completeness sake.
